src/components/NotificationsDrawer/DrawerPanelContent.tsx

```diff
@@ -39,19 +57,139 @@ const EmptyNotifications = () => (
+    </DropdownItem>,
+    <DropdownItem key="unread all" onClick={() => onMarkAllAsUnread()}>
+      Mark visible as unread
+    </DropdownItem>,
+    <Divider key="divider" />,
+    <DropdownItem key="event log">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      View event log
+    </DropdownItem>,
+    <DropdownItem key="notification settings">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      Configure notification settings
+    </DropdownItem>,
+    <DropdownItem key="notification preferences">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      Manage my notification preferences
+    </DropdownItem>,
+  ];
+
+  const filterDropdownItems = () => {
+    const sources = notifications.reduce((acc: string[], { source }) => (acc.includes(source) ? acc : [...acc, source]), []);
+
+    return [
+      <DropdownGroup key="filter-label" label="Show notifications for...">
+        {sources.map((source, index) => (
+          <DropdownItem key={index} onClick={() => onFilterSelect(source)}>
+            <Checkbox isChecked={activeFilters.includes(source)} id={index.toString()} className="pf-u-mr-xs" />
+            {source}
+          </DropdownItem>
+        ))}
+        <Divider />
+        <DropdownItem key="reset-filters" onClick={() => setActiveFilters([])}>
+          <Button variant="link" isInline>
+            Reset filters
+          </Button>
+        </DropdownItem>
+      </DropdownGroup>,
+    ];
+  };
+
+  const renderNotifications = () => {
+    if (notifications.length === 0) {
+      return <EmptyNotifications />;
+    }
+
+    const sortedNotifications = orderBy(
+      filteredNotifications?.length > 0 ? filteredNotifications : notifications,
+      ['read', 'created'],
+      ['asc', 'asc']
+    );
+
+    return sortedNotifications.map((notification, index) => <NotificationItem key={index} notification={notification} />);
+  };
+
   return (
     <NotificationDrawer ref={innerRef}>
-      <NotificationDrawerHeader onClose={() => dispatch(toggleNotificationsDrawer())}>
-        <Button variant="plain" aria-label="Filter">
-          <FilterIcon />
-        </Button>
-        <Button variant="plain" aria-label="Mark as read">
-          <EllipsisVIcon />
-        </Button>
+      <NotificationDrawerHeader onClose={() => onNotificationsDrawerClose()}>
+        {activeFilters.length > 0 && <Badge isRead>{activeFilters.length}</Badge>}
+        <Dropdown
+          toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+            <MenuToggle ref={toggleRef} onClick={() => setIsFilterDropdownOpen(!isFilterDropdownOpen)} id="filter-toggle" isFullWidth variant="plain">
+              <FilterIcon />
+            </MenuToggle>
+          )}
+          isOpen={isFilterDropdownOpen}
+          id="filter-dropdown"
+          aria-label="Notifications filter"
+          isPlain
+        >
+          {filterDropdownItems()}
+        </Dropdown>
+        <Dropdown
```

Comment:
The dropdown is missing the `onOpenChange` prop to properly react to click events. You are also missing the `popperProps` config to ensure the dropdown menu is not hiding behind the edge of the screen.